clients/src/main/java/org/apache/kafka/common/security/authenticator/DefaultKafkaPrincipalBuilder.java

```diff
@@ -188,9 +185,7 @@ public KafkaPrincipal deserialize(byte[] bytes) {
             throw new SerializationException("Invalid principal data version " + version);
```

Comment:
Could we use ```HIGHEST_SUPPORTED_VERSION``` and ```LOWEST_SUPPORTED_VERSION``` to replace origin condition ("0" and ```DefaultPrincipalData.SCHEMAS.length```) ?

Reply:
Good point, addressed.